### PR TITLE
33 dev imgpost service briiite

### LINKED ISSUE #33: [DEV] ImgPost Service

### PR DESCRIPTION
**Pull request by briiitelord**
Updated ImgPostService and added ImgPostServiceImpl class with override methods

### CODE REVIEW

File: src/main/java/com/amplifier/services/ImgPostService.java [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+
 public interface ImgPostService {
 
+    public List<ImgPost> getAllImgPosts();
+
+    public boolean createImgPost(ImgPost imgPost);
+
+    public ImgPost getImgPostById(int Id);
+
+    public boolean updateImgPost(ImgPost mockImgPostModification);
```
Comment (FullSol) [inline src/main/java/com/amplifier/services/ImgPostService.java]: The argument variable should not include mock here. Just "imgPost"

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public boolean createImgPost(ImgPost imgPost) {
+        return imgPostRepository.create(imgPost);
+
+    }
+
+    @Override
+    public ImgPost getImgPostById(int Id) {
+        return null;
```
Comment (FullSol) [inline src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: This should be an actual return, not just null.

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public boolean updateImgPost(ImgPost mockImgPostModification) {
+        return false;
```
Comment (FullSol) [inline src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: This should be an actual return

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public boolean deleteImgPostById(int Id) {
+        return false;
```
Comment (FullSol) [inline src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: This should be an actual return

**Review by FullSol — CHANGES_REQUESTED** [review]
Since you are working on the Service layer and not a test, you would do the full implementation for these files.

**Comment by FullSol** [comment]
I think you may have seen one of my examples with the mock in them. My IDE did that and I did not realize, sorry for the bad example on that. Thanks for getting so much done as well!

File: src/main/java/com/amplifier/services/ImgPostService.java (hunk 1 above)
Comment (briiitelord) [reply src/main/java/com/amplifier/services/ImgPostService.java]: Updated variable - done

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java (hunk 2 above)
Comment (briiitelord) [reply src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: Updated Return - done

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java (hunk 4 above)
Comment (briiitelord) [reply src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: Updated Return - done

File: src/main/java/com/amplifier/services/ImgPostServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public boolean updateImgPost(ImgPost imgPost) {
+        return imgPostRepository.update(imgPost);
+    }
+
+    @Override
+    public boolean deleteImgPostById(int Id) {
+        return imgPostRepository.deleteById(Id);
+    }
+    
+}
```
Comment (pyaeger) [inline src/main/java/com/amplifier/services/ImgPostServiceImpl.java]: Thanks for all the work! FYI: GitHub is giving me a warning icon saying "no new line at end of file" .. guess it wants a new line after that last curly brace.

**Review by pyaeger — APPROVED** [review]

**Review by FullSol — APPROVED** [review]

**Pull request merged**